addons/delivery/models/delivery_carrier.py

```diff
@@ -47,10 +33,39 @@ class DeliveryCarrier(models.Model):
     # Internals for shipping providers #
     # -------------------------------- #
 
-    delivery_type = fields.Selection([('grid', 'Default Delivery')], string="Delivery Type", default='grid', required='True')
-
+    sequence = fields.Integer(help="Determine the display order", default=10)
+    delivery_type = fields.Selection([('fixed', 'Fixed Price'), ('base_on_rule', 'Based on Rules')], string='Price Computation', default='fixed', required=True)
+    product_type = fields.Selection(related='product_id.type', default='service')
+    product_sale_ok = fields.Boolean(related='product_id.sale_ok', default=False)
+    partner_id = fields.Many2one('res.partner', string='Transporter Company', required=True, help="The partner that is doing the delivery service.")
+    product_id = fields.Many2one('product.product', string='Delivery Product', required=True, ondelete="cascade")
     price = fields.Float(compute='get_price')
     available = fields.Boolean(compute='get_price')
+    free_if_more_than = fields.Boolean('Free if Order total is more than', help="If the order is more expensive than a certain amount, the customer can benefit from a free shipping", default=False)
+    amount = fields.Float(string='Amount', help="Amount of the order to benefit from a free shipping, expressed in the company currency")
+    country_ids = fields.Many2many('res.country', 'delivery_carrier_country_rel', 'carrier_id', 'country_id', 'Countries')
+    state_ids = fields.Many2many('res.country.state', 'delivery_carrier_state_rel', 'carrier_id', 'state_id', 'States')
+    zip_from = fields.Char('Zip From')
+    zip_to = fields.Char('Zip To')
+    price_rule_ids = fields.One2many('delivery.price.rule', 'carrier_id', 'Pricing Rules', copy=True)
+    state = fields.Selection([('test', 'Testing'), ('production', 'Production')], 'State')
+    fixed_price = fields.Float(compute='_compute_fixed_price', inverse='_set_product_fixed_price', store=True, string='Fixed Price',help="Keep empty if the pricing depends on the advanced pricing per destination")
+
+    @api.depends('product_id.list_price', 'product_id.product_tmpl_id.list_price')
+    def _compute_fixed_price(self):
+        for carrier in self:
+            carrier.fixed_price = carrier.product_id.list_price
+
+    def _set_product_fixed_price(self):
+        for carrier in self:
+            carrier.product_id.list_price = carrier.fixed_price
+
+    @api.onchange('delivery_type')
+    def onchange_delivery_type(self):
+        if self.delivery_type in ["fixed", "base_on_rule"]:
+            self.state = 'production'
+        else:
+            self.state = 'test'
```

Comment:
hello @dmo-odoo ,
If type is external provider modules default state is going to set Testing in statusbar , user can change production mode by clicking on statusbar, if the type is 'fixed' or 'based ion rule' it's going to set default state to production because there is no state like testing in that,  so status bar is going to set invisible for fixed and based_on_rule. Thanks